src/java.base/share/classes/java/lang/Long.java

```diff
@@ -446,36 +449,111 @@ private static void formatUnsignedLong0UTF16(long val, int shift, byte[] buf, in
     }
 
     static String fastUUID(long lsb, long msb) {
```

Comment:
This method should probably have an assert or something that `COMPACT_STRINGS` is `true`, e.g.:
```suggestion
    static String fastUUID(long lsb, long msb) {
        if (!COMPACT_STRINGS) {
            throw new InternalError("Long::fastUUID requires compact strings");
        }

```

Reply:
it's none-public method, only for UUID.toString

Reply:
now the fastUUID method codeSize is 325, which is equal to the maximum value of FreqInlineSize under aarch64 and x64. if additional assertion is added, the method codeSize > 325, cannot be inlined, resulting in performance degradation.